API tests are failing with new carts image

After fix of carts native image the api tests for carts service started failing. 
```
INFO  ?.o.i.0.3-SNAPSHOT] - Creating container for image: iad.ocir.io/cloudnative-devrel/micronaut-showcase/mushop/carts-graalvm:1.0.3-SNAPSHOT
    06:07:09.359 [Test worker] INFO  ?.o.i.0.3-SNAPSHOT] - Starting container with ID: e8bc3aaa160c68fe88f559a9a4cb539f652ee3d56ae5c54c3dc17ff9067cc4ff
    06:07:09.798 [Test worker] INFO  ?.o.i.0.3-SNAPSHOT] - Container iad.ocir.io/cloudnative-devrel/micronaut-showcase/mushop/carts-graalvm:1.0.3-SNAPSHOT is starting: e8bc3aaa160c68fe88f559a9a4cb539f652ee3d56ae5c54c3dc17ff9067cc4ff
    06:08:09.817 [Test worker] ERROR ?.o.i.0.3-SNAPSHOT] - Could not start container
    java.lang.IllegalStateException: Container exited with code 1
    	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:469)
    	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
    	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
    	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
    	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
    	at api.AbstractDatabaseServiceTest.getProperties(AbstractDatabaseServiceTest.java:38)
    	at api.CartsServiceAnonymousTest.getProperties(CartsServiceAnonymousTest.java:37)
    	at io.micronaut.test.extensions.junit5.MicronautJunit5Extension.resolveTestProperties(MicronautJunit5Extension.java:163)
    	at io.micronaut.test.extensions.junit5.MicronautJunit5Extension.resolveTestProperties(MicronautJunit5Extension.java:47)
    	at io.micronaut.test.extensions.AbstractMicronautExtension.beforeClass(AbstractMicronautExtension.java:214)
    	at io.micronaut.test.extensions.junit5.MicronautJunit5Extension.beforeAll(MicronautJunit5Extension.java:54)
    	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$8(ClassBasedTestDescriptor.java:368)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:368)
    	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:192)
    	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:136)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
    	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
    	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
    	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
    	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
    	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
    	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
    	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
    	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
    	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
    	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
    	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
    	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
    	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
    	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
    	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
    	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
    	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
    	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
    	at com.sun.proxy.$Proxy2.stop(Unknown Source)
    	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:135)
    	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
    	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
    	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
    	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
    	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
    	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
    	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
    	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
    	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
    	at java.base/java.lang.Thread.run(Thread.java:829)
    06:08:09.855 [Test worker] ERROR ?.o.i.0.3-SNAPSHOT] - Log output from the failed container:
     __  __ _                                  _   
    |  \/  (_) ___ _ __ ___  _ __   __ _ _   _| |_ 
    | |\/| | |/ __| '__/ _ \| '_ \ / _` | | | | __|
    | |  | | | (__| | | (_) | | | | (_| | |_| | |_ 
    |_|  |_|_|\___|_|  \___/|_| |_|\__,_|\__,_|\__|
      Micronaut (v2.5.4)

    06:07:13.065 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
    06:07:13.704 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
    06:07:13.869 [main] INFO  m.carts.soda.OracleSodaClientFactory - Initializing DB connection cart
    06:07:13.983 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [oracle.soda.rdbms.OracleRDBMSClient] could not be loaded: Error instantiating bean of type [oracle.soda.rdbms.OracleRDBMSClient]: Error initializing SODA: null
    io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [oracle.soda.rdbms.OracleRDBMSClient] could not be loaded: Error instantiating bean of type [oracle.soda.rdbms.OracleRDBMSClient]: Error initializing SODA: null
    	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1568)
    	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:234)
    	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2907)
    	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:231)
    	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:180)
    	at io.micronaut.runtime.Micronaut.start(Micronaut.java:71)
    	at io.micronaut.runtime.Micronaut.run(Micronaut.java:311)
    	at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
    	at mushop.carts.Application.main(Application.java:30)
    Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [oracle.soda.rdbms.OracleRDBMSClient]: Error initializing SODA: null
    	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2008)
    	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2770)
    	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2756)
    	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2294)
    	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1562)
    	... 8 common frames omitted
    Caused by: io.micronaut.context.exceptions.ConfigurationException: Error initializing SODA: null
    	at mushop.carts.soda.OracleSodaClientFactory.sodaClient(OracleSodaClientFactory.java:103)
    	at mushop.carts.soda.$OracleSodaClientFactory$SodaClient0Definition.doBuild(Unknown Source)
    	at io.micronaut.context.AbstractParametrizedBeanDefinition.build(AbstractParametrizedBeanDefinition.java:118)
    	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:149)
    	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1979)
    	... 12 common frames omitted
    Caused by: oracle.soda.rdbms.impl.SODAUtils$1OracleSQLException: null
    	at oracle.soda.rdbms.impl.SODAUtils.makeExceptionWithSQLText(SODAUtils.java:119)
    	at oracle.soda.rdbms.impl.SODAUtils.makeExceptionWithSQLText(SODAUtils.java:82)
    	at oracle.soda.rdbms.impl.OracleDatabaseImpl.loadCollection(OracleDatabaseImpl.java:1809)
    	at oracle.soda.rdbms.impl.OracleDatabaseImpl.openCollection(OracleDatabaseImpl.java:771)
    	at oracle.soda.rdbms.impl.OracleDatabaseImpl.openCollection(OracleDatabaseImpl.java:730)
    	at mushop.carts.soda.OracleSodaClientFactory.sodaClient(OracleSodaClientFactory.java:77)
    	... 16 common frames omitted
    Caused by: java.sql.SQLException: ORA-06550: line 2, column 3:
    PLS-00201: identifier 'DBMS_SODA_ADMIN' must be declared
    ORA-06550: line 2, column 3:
    PL/SQL: Statement ignored

    	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:628)
    	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:562)
    	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1145)
    	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:726)
    	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:291)
    	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:492)
    	at oracle.jdbc.driver.T4CCallableStatement.doOall8(T4CCallableStatement.java:144)
    	at oracle.jdbc.driver.T4CCallableStatement.executeForRows(T4CCallableStatement.java:1034)
    	at oracle.jdbc.driver.OracleStatement.executeSQLStatement(OracleStatement.java:1507)
    	at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1287)
    	at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3735)
    	at oracle.jdbc.driver.OraclePreparedStatement.execute(OraclePreparedStatement.java:3933)
    	at oracle.jdbc.driver.OracleCallableStatement.execute(OracleCallableStatement.java:4279)
    	at oracle.jdbc.driver.OraclePreparedStatementWrapper.execute(OraclePreparedStatementWrapper.java:1010)
    	at oracle.soda.rdbms.impl.OracleDatabaseImpl.loadCollection(OracleDatabaseImpl.java:1784)
    	... 19 common frames omitted
    Caused by: oracle.jdbc.OracleDatabaseException: ORA-06550: line 2, column 3:
    PLS-00201: identifier 'DBMS_SODA_ADMIN' must be declared
    ORA-06550: line 2, column 3:
    PL/SQL: Statement ignored

    	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:632)
    	... 33 common frames omitted


CartsServiceAnonymousTest > initializationError FAILED
    org.testcontainers.containers.ContainerLaunchException: Container startup failed
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:330)
        at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
        at api.AbstractDatabaseServiceTest.getProperties(AbstractDatabaseServiceTest.java:38)
        at api.CartsServiceAnonymousTest.getProperties(CartsServiceAnonymousTest.java:37)

        Caused by:
        org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
            at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
            at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
            ... 3 more

            Caused by:
            org.testcontainers.containers.ContainerLaunchException: Could not create/start container
                at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:497)
                at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
                at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
                ... 4 more

                Caused by:
                java.lang.IllegalStateException: Container exited with code 1
                    at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:469)
                    ... 6 more

```

--- a/src/api/src/test/java/api/AbstractDatabaseServiceTest.java
+++ b/src/api/src/test/java/api/AbstractDatabaseServiceTest.java
@@ -6,6 +6,9 @@ import io.micronaut.http.HttpResponse;
 import io.micronaut.http.annotation.Post;
 import io.micronaut.http.client.annotation.Client;
 import io.micronaut.test.support.TestPropertyProvider;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
 import org.junit.jupiter.api.AfterAll;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.containers.Network;
@@ -34,6 +37,13 @@ abstract class AbstractDatabaseServiceTest implements TestPropertyProvider {
                 .withNetwork(Network.SHARED)
                 .withNetworkAliases("oracledb");
         oracleContainer.start();
+        try (Connection connection = oracleContainer.createConnection("")) {
+            try (final PreparedStatement ps = connection.prepareStatement("GRANT SODA_APP TO " + oracleContainer.getUsername())) {
+                ps.execute();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Unable to setup SODA: " + e.getMessage(), e);
+        }        
         serviceContainer = initService();
         serviceContainer.start();
         return Map.of(

